nixos/modules/services/video/mirakurun.nix

```diff
@@ -0,0 +1,138 @@
+        };
+
+        nice = mkOption {
+          type = types.ints.between (-20) 19;
+          default = -10;
+          description = "Nice value for Mirakurun.";
+        };
+
+        ioSchedulingClass = mkOption {
+          type = types.enum [ "none" "realtime" "best-effort" "idle" ];
+          default = "realtime";
+          description = "IO scheduling class for Mirakurun.";
+        };
+
+        ioSchedulingPriority = mkOption {
+          type = types.ints.between 0 7;
+          default = 7;
+          description = "IO scheduling priority for Mirakurun.";
+        };
+
+        logLevel = mkOption {
+          type = types.ints.between (-1) 3;
+          default = 2;
+          description = "Mirakurun log level";
+        };
+
+        domainSocket = mkOption {
+          type = with types; nullOr path;
+          default = "/var/run/mirakurun/mirakurun.sock";
+          description = "Path to domain socket used to communicate with Mirakurun.";
+        };
+
+        port = mkOption {
+          type = with types; nullOr ints.u16;
+          default = 40772;
+          description = "Port to listen on.";
+        };
+
+        extraConfig = mkOption {
+          type = types.attrs;
+          default = {};
+          example = {
+            highWaterMark = 25165824;
+            overflowTimeLimit = 30000;
+          };
+          description = "Extra options which are added to server.yml.";
+        };
+
+        tunerConfig = mkOption {
+          type = with types; nullOr (listOf attrs);
+          default = [
+            {
+              name = "dummy-tuner";
+              isDisabled = true;
+            }
+          ];
+          example = [
+            {
+              name = "tuner-name";
+              types = [ "GR" "BS" "CS" "SKY" ];
+              dvbDevicePath = "/dev/dvb/adapterX/dvrX";
+            }
+          ];
+          description = "Options which are added to tuners.yml";
+        };
+      };
+    };
+
+    config = mkIf cfg.enable {
+      environment.systemPackages = [ mirakurun ];
+      environment.etc = {
+        "mirakurun/server.yml".text = builtins.toJSON serverConfig;
+        "mirakurun/tuners.yml".text = builtins.toJSON cfg.tunerConfig;
+      };
+
+      users.extraUsers = singleton {
+        name = cfg.user;
+        group = cfg.group;
+        description = "Mirakurun server";
+      };
+
+      systemd.services.mirakurun = {
+        description = mirakurun.meta.description;
+        wantedBy = [ "multi-user.target" ];
+        after = [ "network.target" ];
+        serviceConfig = {
+          ExecStart = "${mirakurun}/bin/mirakurun";
+          User = cfg.user;
+          Group = cfg.group;
+          PermissionsStartOnly = true;
+          Nice = cfg.nice;
+          IOSchedulingClass = cfg.ioSchedulingClass;
+          IOSchedulingPriority = cfg.ioSchedulingPriority;
+        };
+
+        preStart = ''
+          mkdir -p /var/run/mirakurun
```

Comment:
this can be left to systemd with `RuntimeDirectory=mirakurun` if you prefer. (/var/run is a symlink to /run on my system)